src/utils.ts

```diff
@@ -292,3 +305,36 @@ function validateBuffer(
     throw new Error(`Invalid value: Must be a non-zero ${length}-byte buffer.`);
   }
 }
+
+export const decodeBase58check = (value: string): Buffer => {
+  const base58Check = scureBase58check(sha256);
+
+  try {
+    return Buffer.from(base58Check.decode(value));
+  } catch {
+    throw new Error(
+      `Invalid value: Value is not base58-encoded, or the checksum is invalid.`,
+    );
+  }
+};
+
+export const encodeBase58check = (value: Buffer): string => {
+  const base58Check = scureBase58check(sha256);
+
+  return base58Check.encode(value);
```

Comment:
Can this throw? If so, should we wrap it?

Reply:
AFAIK this just accepts any arbitrary Buffer, so it should never throw.